### PR TITLE
Jarrel/add auth integration test

### PR DESCRIPTION
**Pull request by jarrel-b**
Changes:
- Added a test that creates a new user, logs them in, and then logs them out
- Some refactoring to the test setup and moving methods to functions for reuse

### CODE REVIEW

**Review by bennycio — APPROVED** [review]
So slick! Love that we are using real private keys now for signing and so the tests themselves are so readable. Thanks so much Jarrel!

File: .circleci/config.yml [hunk 1]
```diff
@@ -82,11 +82,32 @@ jobs:
...
+          command: "keybase oneshot"
+      - run:
+          name: "Decrypt secrets"
+          command: keybase decrypt -i _encrypted_internal -o _internal.zip
+      - run:
+          name: "Install cli dependencies"
+          command: sudo apt-get -y install unzip
+      - run:
+          name: "Unzip secrets file"
+          command: echo "A" | unzip _internal.zip
```
Comment (Robinnnnn) [inline .circleci/config.yml:105]: let's goooo

File: .circleci/config.yml [hunk 2]
```diff
@@ -82,11 +82,32 @@ jobs:
...
       - run:
           name: Run integration-tests
           command: |
             mkdir -p /tmp/test-reports
-            gotestsum --junitfile /tmp/test-reports/integration-tests.xml -- -run TestIntegrationTest ./...
+            gotestsum --junitfile /tmp/test-reports/integration-tests.xml -- -run TestIntegrationTest ./server/... -args -chain ethereum -chainID 1
```
Comment (Robinnnnn) [inline .circleci/config.yml:110]: 😍

**Review by Robinnnnn — APPROVED** [review]
LFGGG

**Pull request merged**